### ISSUE TITLE
Swagger component generation issue

### ISSUE THREAD

**Issue by yondersever** [springdoc/springdoc-openapi#679]
Hi,

We have a Person class. Person class has a property with type PersonDetail. And PersonDetail has a property with type Mail class.

When we start the application and navigate to swagger ui html page, Mail class is not generated in components section of openapi definition and we get "Could not resolve reference: Could not resolve pointer: /components/schemas/Mail does not exist in document" error on page. As we checked if there is a complex type in the third level that time springdoc can not resolve that type.
Person and PersonDetail works fine  but Mail fails.

Person->PersonDetail->Mail

public class Person {
    private String name;

    private PersonDetail personDetail;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public PersonDetail getPersonDetail() {
        return personDetail;
    }

    public void setPersonDetail(PersonDetail personDetail) {
        this.personDetail = personDetail;
    }
}

public class PersonDetail {
    private String surname;

    private List<Mail> mails;

    public List<Mail> getMails() {
        return mails;
    }

    public void setMails(List<Mail> mails) {
        this.mails = mails;
    }

}

public class Mail {
    private String mailAddress;

    public String getMailAddress() {
        return mailAddress;
    }

    public void setMailAddress(String mailAddress) {
        this.mailAddress = mailAddress;
    }

}

 @GET(path = "/getPersonTest")
    @Operation(description = "Testttt")
    @ApiResponses(value = { @ApiResponse(responseCode = "200", description = "successful operation",
            content = @Content(schema = @Schema(implementation = Person.class)))})
    public ResponseEntity<Person> getPerson(@RequestParam String name){
        Person person = new Person();
        return ResponseEntity.statusBuilder(HttpStatus.OK).body(person);
    }

**Comment by bnasslahsen** [comment]
@yondersever, 

 When you create an issue. Please make sure you respect code formatting to help us speed up diagnosis.

 You didn't mention the springdoc and the spring-boot version you are using.
 You didn't describe the generated api-docs exactly.


Anyway, your issue is not confirmed. Here is the generated OpenAPI description for the code you have provided.

```yaml
openapi: 3.0.1
info:
  title: OpenAPI definition
  version: v0
servers:
  - url: 'http://localhost:8080'
    description: Generated server url
paths:
  /getPersonTest:
    get:
      tags:
        - hello-controller
      description: Testttt
      operationId: getPerson
      parameters:
        - name: name
          in: query
          required: true
          schema:
            type: string
      responses:
        '200':
          description: successful operation
          content:
            '*/*':
              schema:
                $ref: '#/components/schemas/Person'
components:
  schemas:
    Mail:
      type: object
      properties:
        mailAddress:
          type: string
    Person:
      type: object
      properties:
        name:
          type: string
        personDetail:
          $ref: '#/components/schemas/PersonDetail'
    PersonDetail:
      type: object
      properties:
        mails:
          type: array
          items:
            $ref: '#/components/schemas/Mail'
```

**Comment by mdjimy** [comment]
Hello,

I have exactly the same problem on my project. If I'll follow this example with `Mail`, `Person` and `PersonDetail` then my generated schema looks like this:
```yml
openapi: 3.0.1
info:
  title: OpenAPI definition
  version: v0
servers:
  - url: 'http://localhost:8080'
    description: Generated server url
paths:
  /getPersonTest:
    get:
      tags:
        - hello-controller
      description: Testttt
      operationId: getPerson
      parameters:
        - name: name
          in: query
          required: true
          schema:
            type: string
      responses:
        '200':
          description: successful operation
          content:
            '*/*':
              schema:
                $ref: '#/components/schemas/Person'
components:
  schemas:
    Person:
      type: object
      properties:
        name:
          type: string
        personDetail:
          $ref: '#/components/schemas/PersonDetail'
    PersonDetail:
      type: object
      properties:
        mails:
          type: array
          items:
            $ref: '#/components/schemas/Mail'
```
ie. even thought that it was correctly recognized that `PersonDetail` has a field of type `Mail` and `$ref` is also correctly generated, `Mail` schema is missing.

Spring Boot version: 2.0.9.RELEASE
SpringDoc version: 1.4.0

I've tried various versions of SpringDoc library.
Versions before 1.2.31 "seems" to be working just fine. With version 1.2.31 and newer I'm experiencing this problem.

**Comment by bnasslahsen** [comment]
@mdjimy,

 Please add the sample java code (minimal reproductible) , to help us speed the diagnosis,
 Also, please add the reference of the springdoc-module you are using.

**Comment by mdjimy** [comment]
@bnasslahsen, Thank you very much for very quick response.

I beleave I was able to find root cause of my problem... it seems to be dependency missmatch.
`io.swagger.core.v3:*` (verion 2.1.2) are added to my project transitively via library.
I'm also using `com.atlassian.oai:swagger-request-validator-springmvc:2.8.4`... only for tests, but even so thanks to that maven decided to resolve `swagger-core` version to one the defined by this test dependency.

The result is that all `io.swagger.core.v3` versions are of version 2.1.2 except `swagger-core` which version is 2.0.9. On first sight everything works just fine but in some rare cases some schemas are not generated.

Problem was resolved by adding exclusion of `io.swagger.core.v3:swagger-core` to `swagger-request-validator-springmvc` dependency definition.
